src/Windows/Avalonia.Win32/WindowImpl.cs

```diff
@@ -342,31 +340,25 @@ public void SetInputRoot(IInputRoot inputRoot)
 
         public void Hide()
         {
-            if (_parent != null)
-            {
-                _parent._disabledBy.Remove(this);
-                _parent.UpdateEnabled();
-                _parent = null;
-            }
-
             UnmanagedMethods.ShowWindow(_hwnd, ShowWindowCommand.Hide);
         }
 
         public virtual void Show()
         {
-            SetWindowLongPtr(_hwnd, (int)WindowLongParam.GWL_HWNDPARENT, IntPtr.Zero);
+            SetWindowLongPtr(_hwnd, (int)WindowLongParam.GWL_HWNDPARENT, _parent != null ? _parent._hwnd : IntPtr.Zero);
             ShowWindow(_showWindowState);
         }
 
-        public void ShowDialog(IWindowImpl parent)
+        public Action GotInputWhenDisabled { get; set; }
+
+        public void SetParent(IWindowImpl parent)
         {
             _parent = (WindowImpl)parent;
-            _parent._disabledBy.Add(this);
-            _parent.UpdateEnabled();
-            SetWindowLongPtr(_hwnd, (int)WindowLongParam.GWL_HWNDPARENT, ((WindowImpl)parent)._hwnd);
-            ShowWindow(_showWindowState);
+            SetOwnerHandle(_parent._hwnd);
```

Comment:
This seems to be the only usage, we can call `SetWindowLongPtr` directly, I think.

Reply:
fixed.